app/src/main/java/com/fish/monsters/features/settings/ui/SettingsScreen.kt

```diff
@@ -0,0 +1,138 @@
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.fish.monsters.R
+import com.fish.monsters.common.models.ui.Language
+import com.fish.monsters.common.utils.settings.SettingsGlobalState
+import com.fish.monsters.common.views.CapText
+import com.fish.monsters.common.views.FadingHorizontalDivider
+import com.fish.monsters.common.views.FishSwitch
+import com.fish.monsters.common.views.PreviewContainer
+import com.fish.monsters.common.views.buttons.DangerOutlinedFishButton
+import com.fish.monsters.common.views.buttons.DropdownTextButton
+import com.fish.monsters.common.views.screenContent.ScreenBox
+import com.fish.monsters.core.SettingsUris
+import com.fish.monsters.features.settings.SettingsViewModel
+import com.fish.monsters.features.settings.ui.components.SettingsButtonGridSection
+import com.fish.monsters.features.settings.ui.components.SettingsClearProgressDialog
+import com.fish.monsters.features.settings.ui.components.SettingsRowSection
+import com.fish.monsters.features.settings.ui.components.SettingsSliderSection
+import org.koin.compose.koinInject
+
+
+@Composable
+fun SettingsScreen(viewModel: SettingsViewModel = koinInject()) {
+    val isClearProgressDialogVisible = remember { mutableStateOf(false) }
+    SettingsScreenContent(
+        viewModel.settingsGlobalState.value, isClearProgressDialogVisible
+    ) { viewModel.invokeEvent(it) }
+}
+
+@Composable
+private fun SettingsScreenContent(
+    settingsGlobalState: SettingsGlobalState,
+    isClearProgressDialogVisible: MutableState<Boolean>,
+    invokeEvent: (event: SettingsEvent) -> Unit
+) {
+    val uriHandler = LocalUriHandler.current
+    val context = LocalContext.current
+    ScreenBox(title = stringResource(R.string.settings)) {
+        Column(
+            Modifier
+                .fillMaxSize()
+                .padding(start = 20.dp, end = 20.dp, top = 30.dp),
+            verticalArrangement = Arrangement.spacedBy(20.dp)
+        ) {
+            SettingsRowSection(label = stringResource(R.string.language)) {
+                DropdownTextButton(
+                    items = Language.values().toList(), onItemClicked = {
+                        invokeEvent(SettingsEvent.ChangeLanguageEvent(context, it))
+                    }, defaultButtonText = stringResource(settingsGlobalState.language.titleId)
+                )
+            }
+            SettingsRowSection(label = stringResource(R.string.vibration)) {
+                FishSwitch(checked = settingsGlobalState.vibration, onCheckedChange = {
+                    invokeEvent(SettingsEvent.UpdateSettingsEvent(vibration = it))
+                    invokeEvent(SettingsEvent.SaveCurrentSettings)
+                })
```

Comment:
Kolejna dobra praktyka, czyli nie przesyłanie klasy state dalej (oby nasz widok go zmieniał) tylko kontrolowaniu widoku z naszego ekranu, ułatwia to testowanie a także dzięki temu łatwiej nam użyć potem naszych widoków w innch miejscach.
[Więcej o tym](https://developer.android.com/jetpack/compose/state-hoisting#hoisting-composables)

Reply:
Aj aj captain